lib/iris/fileformats/cf.py

```diff
@@ -1539,19 +1600,42 @@ def _span_check(
         # Determine whether there are any formula terms that
         # may be promoted to a CFDataVariable and restrict promotion to only
         # those formula terms that are reference surface/phenomenon.
-        for cf_var in self.cf_group.formula_terms.values():
-            for cf_root, cf_term in cf_var.cf_terms_by_root.items():
-                cf_root_var = self.cf_group[cf_root]
-                name = cf_root_var.standard_name or cf_root_var.long_name
-                terms = reference_terms.get(name, [])
-                if isinstance(terms, str) or not isinstance(terms, Iterable):
-                    terms = [terms]
-                cf_var_name = cf_var.cf_name
-                if cf_term in terms and cf_var_name not in self.cf_group.promoted:
-                    data_var = CFDataVariable(cf_var_name, cf_var.cf_data)
-                    self.cf_group.promoted[cf_var_name] = data_var
-                    _build(data_var)
-                    break
+        if iris.FUTURE.derived_bounds:
+            for cf_var in self.cf_group.formula_terms.values():
+                if self.cf_group[cf_var.cf_name] is CFBoundaryVariable:
+                    continue
+                else:
+                    for cf_root, cf_term in cf_var.cf_terms_by_root.items():
+                        cf_root_var = self.cf_group[cf_root]
+                        if not hasattr(cf_root_var, "standard_name"):
+                            continue
+                        name = cf_root_var.standard_name or cf_root_var.long_name
```

Comment:
if you reach this point you will have a standard name, so `or cf_root_var.long_name` is unnecessary.